src/test/java/net/snowflake/ingest/streaming/internal/StreamingIngestBigFilesIT.java

```diff
@@ -0,0 +1,199 @@
+    for (int i = 0; i < numChannels; i++) {
+      tableIdToNumChannels.put(
+          i % numTables, tableIdToNumChannels.getOrDefault(i % numTables, 0) + 1);
+    }
+    for (int i = 0; i < numTables; i++) {
+      String tableName = tableNamePrefix + i;
+      createTableForTest(tableName);
+    }
+
+    ingestRandomRowsToTable(
+        tableNamePrefix, numTables, numChannels, batchSize, numBatches, isNullable);
+
+    for (int i = 0; i < numTables; i++) {
+      int numChannelsToTable = tableIdToNumChannels.get(i);
+      verifyTableRowCount(
+          batchSize * numBatches * numChannelsToTable,
+          jdbcConnection,
+          testDb,
+          TEST_SCHEMA,
+          tableNamePrefix + i);
+
+      // select * to ensure scanning works
+      ResultSet result =
+          jdbcConnection
+              .createStatement()
+              .executeQuery(
+                  String.format(
+                      "select * from %s.%s.%s", testDb, TEST_SCHEMA, tableNamePrefix + i));
+      result.next();
+      Assert.assertNotNull(result.getString("STR"));
+    }
+  }
+
+  private void ingestRandomRowsToTable(
+      String tablePrefix,
+      int numTables,
+      int numChannels,
+      int batchSize,
+      int iterations,
+      boolean isNullable)
+      throws ExecutionException, InterruptedException {
+
+    List<Map<String, Object>> rows = new ArrayList<>();
+    for (int i = 0; i < batchSize; i++) {
+      Random r = new Random();
+      rows.add(TestUtils.getRandomRow(r, isNullable));
+    }
+
+    ExecutorService testThreadPool = Executors.newFixedThreadPool(numChannels);
+    CompletableFuture[] futures = new CompletableFuture[numChannels];
+    List<SnowflakeStreamingIngestChannel> channelList = new ArrayList<>();
+    for (int i = 0; i < numChannels; i++) {
+      final String channelName = "CHANNEL" + i;
+      int finalI = i;
+      futures[i] =
+          CompletableFuture.runAsync(
+              () -> {
+                int targetTable = finalI % numTables;
+                SnowflakeStreamingIngestChannel channel =
+                    openChannel(tablePrefix + targetTable, channelName);
+                channelList.add(channel);
+                for (int val = 0; val < iterations; val++) {
+                  TestUtils.verifyInsertValidationResponse(
+                      channel.insertRows(rows, Integer.toString(val)));
+                }
+              },
+              testThreadPool);
+    }
+    CompletableFuture joined = CompletableFuture.allOf(futures);
+    joined.get();
+    channelList.forEach(channel -> TestUtils.waitChannelFlushed(channel, iterations));
+    testThreadPool.shutdown();
+  }
+
+  private void createTableForTest(String tableName) {
+    try {
+      jdbcConnection
+          .createStatement()
+          .execute(
+              String.format(
+                  "create or replace table %s (\n"
+                      + "                                    num_2_1 NUMBER(2, 1),\n"
```

Comment:
I would test it for each data type separately.

Reply:
This is probably not urgent but more relevant if we start to rely on any byte size again. Especially after revisiting Parquet size calculation.

Reply:
Agree.